qcodes/dataset/data_set_protocol.py

```diff
@@ -2,6 +2,9 @@
 
 import sys
 
+from dask.diagnostics.progress import ProgressBar
```

Comment:
Add dask requirement. Should this be optional?

Reply:
Yes that seems ok. Note that could could also depend on xarray[parallel] to pull in dask but that installs dask with the complete subtarget which has even more dependencies